modules/exploits/multi/http/struts2_namespace_ognl.rb

```diff
@@ -0,0 +1,337 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::EXE
+
+  # Eschewing CmdStager for now, since the use of '\' and ';' are killing me
+  #include Msf::Exploit::CmdStager   # https://github.com/rapid7/metasploit-framework/wiki/How-to-use-command-stagers
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Apache Struts2 Namespace RCE',
+      'Description'    => %q{
+        This module exploits a remote code execution vulnerability in Apache Struts
+        version 2.3 - 2.3.4, and 2.5 - 2.5.16. Remote Code Execution can be performed
+        via an endpoint that makes use of a redirect action.
+
+        Native payloads will be converted to executables and dropped in the
+        server's temp dir. If this fails, try a cmd/* payload, which won't
+        have to write to the disk.
+      },
+      #TODO: Is that second paragraph above still accurate?
+      'Author'         => [
+        'Man Yue Mo', # Discovery
+        'hook-s3c',   # PoC
+        'asoto-r7',   # Metasploit module
+        'wvu'         # Metasploit module
+      ],
+      'References'     => [
```

Comment:
Please link to https://github.com/hook-s3c/CVE-2018-11776-Python-PoC. Thanks.

Reply:
Sure thing!  I gave @hook-s3c a shout out in the author field and documentation, but missed the link here.